### DISCUSSION TITLE
How should I export separate STL files for gray matter or white matter after creating the head model?

### DISCUSSION THREAD

**Discussion by lingoodluck** [simnibs/simnibs#451] [2. General]

**Comment by lingoodluck** [comment]
import simnibs
from simnibs import mesh_io

head_model : simnibs.Msh = simnibs.read_msh('path/to/file.msh')

tissue_tag = 1001 
head_model = head_model.crop_mesh(tags=[tissue_tag])

mesh_io.write_stl(head_model, 'test.stl')

I exported the STL files for the scalp, skull, CSF, GM, and WM sequentially using this method, but when I import them into COMSOL, it shows that it cannot generate solid objects.

**Reply by TorgeW** [reply]
This is because the surfaces are not watertight. 
If you look at the surfaces you exported in any software, you will see that the surfaces are not closed (WM surface might be closed). 

To get closed surface, first you would need to define what you want the surfaces to be. 
You might then be able to generate surfaces based on the tetrahedra triangles that interface between different tissues.

If you open your charm mesh (.msh file) and you look at the surfaces, you can look at the volume representation compared to the surfaces.

**Reply by lingoodluck** [reply]
Thank you for your answer. I am now facing a new problem. I want to generate a .npy file that represents the electric field magnitude in a specified shape. Is there any way to do this?

**Reply by TorgeW** [reply]
https://github.com/simnibs/simnibs/discussions/453

**Comment by 784622315** [comment]
Hello, I would like to ask about the steps you said to export a closed surface, does it have a specific code？

**Reply by TorgeW** [reply]
This code will add closed surfaces for every tag to the mesh:

```
m = mesh_io.read_msh('mesh.msh')
m.reconstruct_surfaces()
```
